cmd/zt_generic_traverser_test.go

```diff
@@ -38,6 +40,43 @@ type genericTraverserSuite struct{}
 
 var _ = chk.Suite(&genericTraverserSuite{})
 
+// Test follow symlink functionality
+func (s *genericTraverserSuite) TestWalkWithSymlinks(c *chk.C) {
```

Comment:
We also need approx two additional tests, to test the seenPaths functionality.

e.g. a test where tmpDir contains two symlinks to the same target (Test: can we reject duplicates when both are the target of symlinks?)
and a test where symlinkDir contains a link back to a file in tempDir (Test: can we reject duplicates where one is the target of an symlink and the other is not)

Reply:
And it wouldn't hurt to have an test that actually sets up an cycle - eg. tmpDir links to smlinkDir, and smlinxDir contains a link right back to tempDir itself.  If theory (I think) the other two tests I've suggested would cover that, but in practice it won't hurt to explicitly test the cycle case (since that would probably be the most embrassing bug if it went wrong in the wild)

Reply:
Filled out these tests, and added another one actually. Had symlinks target both a root and child directory to ensure deduping occurred there.